### PR TITLE
KOGITO-3997 - Introducing SW Temperature Conversion Example

### PR DESCRIPTION
**Pull request by ricardozanini**
Signed-off-by: Ricardo Zanini <zanini@redhat.com>

See: https://issues.redhat.com/browse/KOGITO-3997

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

**WARNING! Please make sure you are opening your PR against `master` branch!**

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: serverless-workflow-functions-quarkus/pom.xml [hunk 1]
```diff
@@ -37,6 +37,10 @@
       <groupId>org.kie.kogito</groupId>
       <artifactId>kogito-rest-workitem</artifactId>
     </dependency>
+    <dependency>
```
Comment (fjtirado) [inline serverless-workflow-functions-quarkus/pom.xml:40]: Just a question for increasing my knowledge, why is this dependency needed?

Comment (ricardozanini) [reply serverless-workflow-functions-quarkus/pom.xml:40]: Sorry, I missed that one, and I've changed the file. What was the dependency?

**Comment by ricardozanini** [comment]
Depends on https://github.com/kiegroup/kogito-runtimes/pull/942

**Comment by ricardozanini** [comment]
Jenkins please test this

File: serverless-workflow-temperature-conversion/conversion-workflow/pom.xml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      <artifactId>kogito-rest-workitem</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-vertx</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-smallrye-openapi</artifactId>
+    </dependency>
```
Comment (fjtirado) [inline serverless-workflow-temperature-conversion/conversion-workflow/pom.xml:43]: did you try without this one? I suspect it will work

Comment (ricardozanini) [reply serverless-workflow-temperature-conversion/conversion-workflow/pom.xml:43]: This one is needed to provide the OpenAPI (`swagger-ui`) context. I have tried in the past, but without this dependency, it didn't work. Something has changed then?

File: serverless-workflow-temperature-conversion/conversion-workflow/pom.xml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    <groupId>org.kie.kogito</groupId>
+    <version>2.0.0-SNAPSHOT</version>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>conversion-workflow</artifactId>
+  <name>Kogito Example :: Serverless Workflow Temperature Conversion :: Conversion Service</name>
+
+  <properties>
+    <maven.compiler.source>11</maven.compiler.source>
```
Comment (cristianonicolai) [inline serverless-workflow-temperature-conversion/conversion-workflow/pom.xml]: @ricardozanini _maven.compiler.release_ is already set in the root pom do we still need to set this here? if so, maybe move to root pom?

File: serverless-workflow-temperature-conversion/conversion-workflow/src/test/java/org/kie/kogito/serverless/ConversationFlowTest.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    void sanityVerification() {
+        given()
+                .contentType(ContentType.JSON)
+                .when()
+                .body(
+                        Collections
+                                .singletonMap(
+                                        "workflowdata",
+                                        Collections.singletonMap("fahrenheit", "100")))
+                .post("/fahrenheit_to_celsius?forceSync=true")
```
Comment (cristianonicolai) [inline serverless-workflow-temperature-conversion/conversion-workflow/src/test/java/org/kie/kogito/serverless/ConversationFlowTest.java]: is _forceSync=true_ still needed?

File: serverless-workflow-temperature-conversion/multiplication-service/pom.xml [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+      <artifactId>rest-assured</artifactId>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-maven-plugin</artifactId>
+        <version>${version.io.quarkus}</version>
```
Comment (cristianonicolai) [inline serverless-workflow-temperature-conversion/multiplication-service/pom.xml]: plugins should already have their manged version in parent pom

File: serverless-workflow-temperature-conversion/pom.xml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    <artifactId>kogito-examples</artifactId>
+    <version>2.0.0-SNAPSHOT</version>
+  </parent>
+
+  <name>Kogito Example :: Serverless Workflow Temperature Conversion</name>
+  <artifactId>serverless-workflow-temperature-conversion</artifactId>
+  <packaging>pom</packaging>
+
+  <properties>
+    <compiler-plugin.version>3.8.1</compiler-plugin.version>
```
Comment (cristianonicolai) [inline serverless-workflow-temperature-conversion/pom.xml]: these should be already defined in the root pom

File: serverless-workflow-temperature-conversion/conversion-workflow/pom.xml (hunk 2 above)
Comment (fjtirado) [reply serverless-workflow-temperature-conversion/conversion-workflow/pom.xml:43]: Thanks for the info, I did not know was needed for swagger

**Review by fjtirado — APPROVED** [review]
Great work Ricardo!!!

**Comment by ricardozanini** [comment]
@cristianonicolai I believe I covered all your comments, can you take a second look?

**Review by cristianonicolai — APPROVED** [review]

**Comment by ricardozanini** [comment]
jenkins please retest this

**Comment by ricardozanini** [comment]
```
[INFO] [ERROR] Failed to execute goal on project integration-tests-springboot-kakfa-it: Could not resolve dependencies for project org.kie.kogito:integration-tests-springboot-kakfa-it:jar:2.0.0-SNAPSHOT: The following artifacts could not be resolved: ch.qos.logback:logback-classic:jar:1.2.3, com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.11.0, javax.xml.bind:jaxb-api:jar:2.3.1: Failure to find ch.qos.logback:logback-classic:jar:1.2.3 in file:///home/jenkins/.m2/repository was cached in the local repository, resolution will not be reattempted until the update interval of local.mirror has elapsed or updates are forced -> [Help 1]
```

jenkins retest this please

**Comment by tsurdilo** [comment]
@ricardozanini overall looks good to me

**Pull request merged**